### ISSUE TITLE
When is `The last job failed with the following error:` banner shown on workflow page?

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal-web#443]
I love how with the new workflow page, we stay on that page more often and directly see the statuses. Also being able to see errors directly on the page with the `The last job failed with the following error:` banner is super helpful!

We should clarify a bit when the banner is shown though. But it could be that this is resolved on its own by https://github.com/fractal-analytics-platform/fractal-server/issues/1083

=> currently not actionable, but I collect observations here.

------

When running a task multiple times and I had an error run 7 (failed for MIP), I see its error (=> great).
I then rerun the workflow only from Cellpose onwards (running job 8), it finishes successfully. I still get the same banner saying my last job failed. That's not true. But the error is potentially relevant. It saying the last job failed is confusing though.

<img width="1519" alt="Screenshot 2024-04-11 at 17 10 56" src="https://github.com/fractal-analytics-platform/fractal-web/assets/18033446/e0b51149-6ce8-4341-aea0-8c0a5a259b9c">

The corresponding job list when the screenshot above was taken:
<img width="1519" alt="Screenshot 2024-04-11 at 17 11 23" src="https://github.com/fractal-analytics-platform/fractal-web/assets/18033446/47ce8c19-a01e-43e1-92c0-9d5af457a702">

**Comment by jluethi** [comment]
The banner does go away though when I successfully rerun the whole workflow:
<img width="1519" alt="Screenshot 2024-04-11 at 17 24 31" src="https://github.com/fractal-analytics-platform/fractal-web/assets/18033446/887f3f53-fa52-4935-8329-3892e0a33815">

=> maybe we just need to adapt the wording?

I still think it's somewhat confusing to see this banner after running a successful partial workflow though. I wonder whether it should not show up by default, but the red x on the failed task should be clickable to get to the log / to the entry in the workflow page?

**Comment by jluethi** [comment]
I tried to review it with the new status handling. The original flow doesn't work anymore, because that's a restart of a workflow (=> new dataset).

When I run a different workflow with 2 cellpose tasks & the second one fails, I then rerun just the first one. This results in a status page as I would expect it

<img width="1644" alt="Screenshot 2024-04-24 at 17 37 18" src="https://github.com/fractal-analytics-platform/fractal-web/assets/18033446/7c61cf24-7376-4cef-815e-7726c5caf035">

There is a different edge-case that is out-of-scope for the moment that I've documented here: https://github.com/fractal-analytics-platform/fractal-web/issues/471
